pkg/validate/container.go

```diff
@@ -109,6 +112,40 @@ var _ = framework.KubeDescribe("Container", func() {
 			testExecSync(rc, containerID)
 		})
 	})
+
+	Context("runtime should support adding volume and device", func() {
+		var podID string
+		var podConfig *runtimeapi.PodSandboxConfig
+
+		BeforeEach(func() {
+			podID, podConfig = createPodSandboxForContainer(rc)
+			pullImageList(ic, []string{defaultContainerImage})
+		})
+
+		AfterEach(func() {
+			By("stop PodSandbox")
+			rc.StopPodSandbox(podID)
+			By("delete PodSandbox")
+			rc.RemovePodSandbox(podID)
+		})
+
+		It("runtime should support starting container with volume", func() {
+			By("create host path and flag file")
+			hostPath, flagFile := createHostPath(podID)
+
+			defer os.RemoveAll(hostPath) // clean up the TempDir
+
+			By("create container with volume")
+			containerID := createVolumeContainer(rc, "container-with-volume-test-", podID, podConfig, hostPath, flagFile)
+
+			By("test start container with volume")
+			testStartContainer(rc, containerID)
+
+			By("test container exit code")
+			status := getContainerStatus(rc, containerID)
```

Comment:
You may want to use `Eventually` to wait for the container to finish running.

See https://onsi.github.io/gomega/#eventually